app/lib/ResultContext.php

```diff
@@ -231,19 +231,21 @@ public function isNewSearch($pb_is_new_search=null) {
 	 * @return integer - number of page; the first page is 1 (*not* zero). Returns 1 as default if no page has been set for the current context.
 	 */
 	public function getCurrentResultsPageNumber() {
-		if ($pn_page = ($this->opo_request ? intval($this->opo_request->getParameter('page', pString, ['forcePurify' => true])) < 1 : false)) {
-			// no page in request so fetch from context 
-			if ($va_context = $this->getContext()) {
-				$pn_page = $va_context['page'] ?? 1;
+		if ($this->opo_request) {
+			if (($pn_page = intval($this->opo_request->getParameter('page', pString, ['forcePurify' => true]))) < 1) {
+				// no page in request so fetch from context
+				if ($va_context = $this->getContext()) {
+					$pn_page = $va_context['page'] ?? 1;
+				}
+				if (!$pn_page) { $pn_page = 1; }
+				return $pn_page;
+			} else {
```

Comment:
@kehh I originally didn't want to mess with this logic too much, but if we have to change it this significantly to get it to work I think it might make sense to rewrite these entirely. I _think_ we should be getting into the `if ` case when `$this->opo_request` is unset, to either set the value to one from  context or default to 1.